### ISSUE TITLE
build failure in apiMappings on JDK 9

### ISSUE THREAD

**Issue by SethTisue** [playframework/playframework#8555]
seen at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-jdk9-integrate-community-build/338/

```
[play-core] java.lang.NullPointerException
[play-core] 	at BuildSettings$$anonfun$playCommonSettings$17.apply(BuildSettings.scala:118)
[play-core] 	at BuildSettings$$anonfun$playCommonSettings$17.apply(BuildSettings.scala:116)
[play-core] 	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
[play-core] 	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
[play-core] 	at sbt.std.Transform$$anon$4.work(System.scala:63)
[play-core] 	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
[play-core] 	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
[play-core] 	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[play-core] 	at sbt.Execute.work(Execute.scala:237)
[play-core] 	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
[play-core] 	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
[play-core] 	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[play-core] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
[play-core] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[play-core] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
[play-core] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[play-core] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
[play-core] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
[play-core] 	at java.base/java.lang.Thread.run(Thread.java:844)
[play-core] [error] (Play-Streams/*:apiMappings) java.lang.NullPointerException
[play-core:error] java.lang.RuntimeException: 
[play-core:error] 	at scala.sys.package$.error(package.scala:27)
[play-core:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner.com$typesafe$dbuild$support$sbt$SbtRunner$$processCommand(SbtRunner.scala:94)
[play-core:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1$$anonfun$apply$2$$anonfun$apply$3.apply(SbtRunner.scala:75)
[play-core:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1$$anonfun$apply$2$$anonfun$apply$3.apply(SbtRunner.scala:75)
[play-core:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1.apply(SbtRunner.scala:75)
[play-core:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1.apply(SbtRunner.scala:66)
[play-core:error] 	at sbt.IO$.withTemporaryFile(IO.scala:389)
[play-core:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner.run(SbtRunner.scala:66)
[play-core:error] 	at com.typesafe.dbuild.support.sbt.SbtBuilder$.buildSbtProject(SbtBuildRunner.scala:93)
[play-core:error] 	at com.typesafe.dbuild.support.sbt.SbtBuildSystem.runBuild(SbtBuildSystem.scala:96)
[play-core:error] 	at com.typesafe.dbuild.support.sbt.SbtBuildSystem.runBuild(SbtBuildSystem.scala:22)
```

this error was only recently uncovered now that core Play's dependencies all became green in the 2.12.x/JDK9 build

perhaps this is a problem that's already been solved at https://github.com/ThoughtWorksInc/sbt-api-mappings ...?

**Comment by yduman** [comment]
build is also failing with

```
java 10.0.2 2018-07-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.2+13)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.2+13, mixed mode)
```

same error message as above.

Had to rebuild with

```
java version "1.8.0_191"
Java(TM) SE Runtime Environment (build 1.8.0_191-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.191-b12, mixed mode)
```

to make it work.

**Comment by dinoboy197** [comment]
Also failing for me with Java 10 under OpenJDK:
```
openjdk 10.0.2 2018-07-17
OpenJDK Runtime Environment (build 10.0.2+13-Ubuntu-1ubuntu0.18.04.3)
OpenJDK 64-Bit Server VM (build 10.0.2+13-Ubuntu-1ubuntu0.18.04.3, mixed mode)
```

**Comment by dwijnand** [comment]
Should be fixed now with #8918 and #8970.
